src/physics_swe.c

```diff
@@ -293,61 +339,56 @@ static PetscErrorCode RHSFunctionForInternalEdges(RDy rdy, Vec F, CourantNumberD
   PetscCall(VecRestoreArray(rdy->X_local, &x_ptr));
   PetscCall(VecRestoreArray(F, &f_ptr));
 
+  PetscCall(RiemannDataSWEDestroy(datal));
+  PetscCall(RiemannDataSWEDestroy(datar));
+
   PetscFunctionReturn(0);
 }
 
-// applies a reflecting boundary condition on the given boundary, computing
-// fluxes F for the solution vector components X
-static PetscErrorCode ApplyReflectingBC(RDy rdy, RDyBoundary *boundary, PetscReal tiny_h, CourantNumberDiagnostics *courant_num_diags, PetscReal *X,
-                                        PetscReal *F) {
+// Before computing BC fluxes, perform common precomputation irrespective of BC type that include:
+// (i) extracting h/hu/hv from the solution vector X, and
+// (ii) compute velocities (u/v) from momentum (hu/hv).
+static PetscErrorCode PerformPreComputationForBC(RDy rdy, RDyBoundary *boundary, PetscReal tiny_h, PetscInt N, RiemannDataSWE *data, PetscReal cn[N],
```

Comment:
What do you think about renaming this to `PrecomputeBC`? Again, not a big deal. :-)

Reply:
done.